v2/tools/generator/internal/astmodel/renaming_visitor_test.go

```diff
@@ -60,7 +60,7 @@ func TestRenamingVisitor_RewritesResourceOwner(t *testing.T) {
 	defs := make(TypeDefinitionSet)
 	defs.AddAll(badObject, childDef)
 
-	newName := badObject.Name().WithName("GoodName")
+	newName := badObject.Name().WithName("GoodName").(InternalTypeName)
```

Comment:
Hmm, slightly unfortunate but I guess not a big deal.

Reply:
This cast goes away in one of the later PRs in this sequence, when `WithName()` starts returning `InternalTypeName` directly.